### PR TITLE
 CCS-3480: Show assemblies on module detail page

### PR DESCRIPTION
**Pull request by aprajshekhar**
This PR:

1. Introduces a new API endpoint /module/assemblies.json for retrieving list of assembly details corresponding to module variant uuid.
2. Refactors logic for retrieving list of assembly details corresponding to module into a set of utility methods.
3. Adds assembly list to the module details page where each assembly title links to assembly details page.

### CODE REVIEW

File: pantheon-bundle/frontend/src/app/versions.tsx [hunk 1]
```diff
@@ -19,6 +30,7 @@ export interface IProps {
     variant: string
     variantUUID: string
     attributesFilePath: string
+    assemblies?: any
```
Comment (rednitish) [inline pantheon-bundle/frontend/src/app/versions.tsx:33]: ? - is it a typo?

File: scripts/deploy_local.sh [hunk 2]
```diff
@@ -21,7 +21,7 @@ cd $PANTHEON_CODEBASE
 set -e
 # build the distribution
 # build the karaf-features and karaf-config along with pantheon modules
-./mvnw clean install -U -pl sling-org-apache-sling-karaf-features,sling-org-apache-sling-karaf-configs,pantheon-bundle,pantheon-karaf-feature,pantheon-karaf-dist
+./mvnw clean install -U -e  -pl sling-org-apache-sling-karaf-features,sling-org-apache-sling-karaf-configs,pantheon-bundle,pantheon-karaf-feature,pantheon-karaf-dist
```
Comment (rednitish) [inline scripts/deploy_local.sh:24]: Do we really need to change current build command ?

File: pantheon-bundle/frontend/src/app/moduleDisplay.tsx [hunk 3]
```diff
@@ -211,7 +216,7 @@ class ModuleDisplay extends Component<any, any, any> {
         fetch(path + '/en_US.harray.4.json')
             .then(response => response.json())
             .then(responseJSON => {
-                // console.log('fetch results:', responseJSON)
+                 console.log('fetch results:', responseJSON)
```
Comment (rednitish) [inline pantheon-bundle/frontend/src/app/moduleDisplay.tsx:219]: remove all the debug logs.!

File: pantheon-bundle/frontend/src/app/versions.tsx (hunk 1 above)
Comment (aprajshekhar) [reply pantheon-bundle/frontend/src/app/versions.tsx:33]: As discussed, no.

File: scripts/deploy_local.sh (hunk 2 above)
Comment (aprajshekhar) [reply scripts/deploy_local.sh:24]: Reverted the change.

**Review by rednitish — APPROVED** [review]
LGTM

**Pull request merged**